router/broker.go

```diff
@@ -33,12 +35,38 @@ type subscription struct {
 	subscribers map[*wamp.Session]struct{}
 }
 
+// storedEvent is a wrapper around wamp event message with timestamp
+// to be used in event history
+type storedEvent struct {
+	timestamp    time.Time
+	Subscription wamp.ID
+	Publication  wamp.ID
+	Details      wamp.Dict
+	Arguments    wamp.List `wamp:"omitempty"`
+	ArgumentsKw  wamp.Dict `wamp:"omitempty"`
+}
+
+type eventHistoryItem struct {
+	publication *wamp.Publish
+	event       *storedEvent
+}
+
+type eventHistoryStoreItem struct {
+	events         []*eventHistoryItem
+	matchPolicy    string
+	limit          int
+	isLimitReached bool
```

Comment:
Do you need to store the fact that the limit was reached and some things were lost so that this is relayed to the clients?

Reply:
Slice length is not something that is calculated, it is stored in the slice itself. I would remove this field unless there's an API from the client to ask if history is lost.